lib/puppet/provider/aix_object.rb

```diff
@@ -346,7 +346,11 @@ def addcmd(attributes)
   end
 
   def deletecmd
-    [self.class.command(:delete)] + ia_module_args + [@resource[:name]]
+    if self.class == Puppet::Type::User::ProviderAix
```

Comment:
`High-level modules should not depend on low-level modules`

https://en.wikipedia.org/wiki/Dependency_inversion_principle